### ISSUE TITLE
Authenticate/Pass Auth to API 

### ISSUE THREAD

**Issue by coultergeist** [ActoKids/AD440_W19_CloudPracticum#8]
Yitong:
Authenticate/Pass Auth to API 
IAM permissions, permissions to access API based on user, Admin, Super Admin (Identity and Access Management)

Projected Time:
Actual Time:

**Comment by bradleyshao** [comment]
I think this task will take about 10 hours.

**Comment by bradleyshao** [comment]
Introduction video about AWS IAM https://www.aws.training/learningobject/video?id=16448.

**Comment by bradleyshao** [comment]
[AWS IAM features.docx](https://github.com/ActoKids/AD440_W19_CloudPracticum/files/2766604/AWS.IAM.features.docx)

**Comment by bradleyshao** [comment]
AWS IAM features:
1.Enhanced security
IAM enables security best practices by allowing you to grant unique security credentials to users and groups to specify which AWS service APIs and resources they can access. IAM is secure by default; users have no access to AWS resources until permissions are explicitly granted.

2. Granular control
IAM provides the granularity to control a user’s access to specific AWS services and resources using permissions. For example, terminating EC2 instances or reading the contents of an Amazon S3 bucket.

3. Temporary credentials
In addition to defining access permissions directly to users and groups, IAM lets you create roles. Roles allow you to define a set of permissions and then let authenticated users or EC2 instances assume them, getting temporary access to the resources you define.

4. Flexible security credential management
IAM allows you to authenticate users in several ways, depending on how they want to use AWS services. You can assign a range of security credentials including passwords, key pairs, and X.509 certificates. You can also enforce multi-factor authentication (MFA) on users who access the AWS Management Console or use APIs.

**Comment by bradleyshao** [comment]
Any AWS customer can use IAM with no extra charge.

**Comment by bradleyshao** [comment]
Did some research about google OAuth today.

**Comment by bradleyshao** [comment]
<img width="954" alt="default" src="https://user-images.githubusercontent.com/34496812/51504177-79f55f80-1d94-11e9-868d-5353db050772.PNG">

**Comment by bradleyshao** [comment]
<img width="960" alt="2" src="https://user-images.githubusercontent.com/34496812/51504236-b923b080-1d94-11e9-93c2-6b3c2413faf9.PNG">

**Comment by bradleyshao** [comment]
Successfully add the Google OAuth2 to the test page. It allow user to use their google account to login. If login success, user can find the following information in console:  Google ID, User name, User's profile photo, User's email, and the message from Jon's API.

**Comment by bradleyshao** [comment]
[GAuthLoginTest.zip](https://github.com/ActoKids/AD440_W19_CloudPracticum/files/2780826/GAuthLoginTest.zip)

gauthlogintest.html

**Comment by bradleyshao** [comment]
<img width="956" alt="default" src="https://user-images.githubusercontent.com/34496812/51570272-81c80900-1e53-11e9-8757-beef4a2571c3.PNG">

**Comment by bradleyshao** [comment]
<img width="960" alt="2" src="https://user-images.githubusercontent.com/34496812/51570293-868cbd00-1e53-11e9-9e70-ee46cb38411c.PNG">

**Comment by bradleyshao** [comment]
Successfully add the Facebook auth to the test page. It allow user to use their facebook account to login. If login success, user can find the following information in console: Login status. And it will output the message from Jon's API on screen.

**Comment by bradleyshao** [comment]
[fbtest.zip](https://github.com/ActoKids/AD440_W19_CloudPracticum/files/2784877/fbtest.zip)
fbtest.html

**Comment by bradleyshao** [comment]
Projected Time: 10 hrs
Actual Time: 8 hrs
First, I did lots of research about Facebook OAuth and Google OAuth2. Both Facebook and Google provide tutorials about how to setup in facebook app dashboard and google api console with the developer account. I added Erik's function to both test pages, if user successfully use their facebook or google account login, the message from Jon's API will show up.
